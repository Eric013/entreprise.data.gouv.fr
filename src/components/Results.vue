<template>
  <div class="container">
    <loader v-if="resultsAreLoading"></loader>
    <server-error v-else-if="serverError"></server-error>
    <template v-else>
      <results-sirene />
      <hr />
      <results-rna />
      <!-- paginate module on RNA and Sirene, so we use the bigger values -->
      <results-paginate-module :totalPages=biggerNumberPages></results-paginate-module>
    </template>
  </div>
</template>

<script>
import Loader from '@/components/modules/Loader'
import ServerError from '@/components/modules/ServerError'
import ResultsSirene from '@/components/results/ResultsSirene'
import ResultsRNA from '@/components/results/ResultsRNA'
import ResultsPaginateModule from '@/components/results/ResultsPaginateModule'

export default {
  name: 'Results',
  metaInfo () {
    return {
      title: `Recherche : ${this.storedLastFullText()}`
    }
  },
  components: {
    'Loader': Loader,
    'ServerError': ServerError,
    'ResultsSirene': ResultsSirene,
    'ResultsRna': ResultsRNA,
    'ResultsPaginateModule': ResultsPaginateModule
  },
  computed: {
    resultsAreLoading () {
      return this.$store.getters.isFullTextLoading
    },
    serverError () {
      return this.$store.getters.allAPIError500
    },
    biggerNumberPages () {
      return Math.max(this.$store.getters.totalPageNumberSirene, this.$store.getters.totalPageNumberRNA)
    }
  },
  methods: {
    storedLastFullText () {
      return this.$store.state.searchFullText.storedLastFullText
    }
  }
}

</script>

<style lang="scss" scoped>

  .container {
    padding-top: 2em;
    padding-bottom: 2em;
  }

</style>
