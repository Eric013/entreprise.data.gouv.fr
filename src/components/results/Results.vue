<template>
  <div v-if="isSearchNotEmpty" class="justify-content-center">
    <h2 class="justify-content-center">Résultats ({{numberResults}}) :</h2>
    <paginate-module></paginate-module>
    <ul>
      <li v-for="result in storedResultsEtablissements">
        <router-link :to="{ name: 'Entreprise', params: { siret: result['siret']}}">
          {{ result['nom_raison_sociale'] }} ({{ result['code_postal'] }})
        </router-link>
      </li>
    </ul>
    <p v-if="numberResults === 0">Aucun résultat trouvé</p>
    <!-- {{debugSeeAllEtablissements}} -->
    <!-- {{debugSeeResponseStatus}} -->
  </div>
</template>

<script>
import store from '@/store/store.js'
import PaginateModule from '@/components/paginate/PaginateModule.vue'

export default {
  name: 'Results',
  components: {
    'PaginateModule': PaginateModule
  },
  computed: {
    isSearchNotEmpty: function () {
      return store.state.storedFullText !== ''
    },
    storedResultsEtablissements () {
      return store.getters.storedResultsEtablissements
    },
    numberResults () {
      return store.getters.numberResults
    // },
    // debugSeeAllEtablissements () {
    //   return store.state.storedResults
    // },
    // debugSeeResponseStatus () {
    //   return store.state.storedStatus
    }
  }
}

</script>
