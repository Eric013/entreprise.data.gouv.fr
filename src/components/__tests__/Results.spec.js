import Results from '@/components/Results.vue'
import { createLocalVue, shallow, mount, RouterLinkStub } from '@vue/test-utils'
import { __createMocks as createStoreMocks } from '@/store/index.js'
import Vuex from 'vuex'

const localVue = createLocalVue()
localVue.use(Vuex)
jest.mock('@/store/index.js')

describe('Results.vue', () => {
  let storeMocks
  let wrapperResults
  let results
  let $router
  
  beforeEach(() => {
    $router = {
      params: 'mock-params',
      push: jest.fn()
    }

    storeMocks = createStoreMocks()
    
    wrapperResults = shallow(Results, {
      localVue,
      store: storeMocks.store,
      stubs: ['router-link', 'router-view'],
      mocks: { $router }
    })
    results = wrapperResults.vm
  })

  test('Computed value isSearchNotEmpty check if search is not empty', () =>{
    storeMocks.store.state.search.storedFullText = ''
    expect(results.isSearchNotEmpty).toBeFalsy()
    storeMocks.store.state.search.storedFullText = 'placeholder'
    expect(results.isSearchNotEmpty).toBeTruthy()
  })

  test('Computed value storedResultsEtablissements return the right getter', () => {
    expect(results.storedResultsEtablissements).toBe(storeMocks.store.getters.storedResultsEtablissements)
  })

  test('Computed value NumberResult return the right getter', () => {
    expect(results.numberResults).toBe(storeMocks.store.getters.numberResults)
  })

  jest.mock('@/store/index.js', () => ({
    getters: {
      numberResults: jest.fn().mockReturnValue(0)
    }
  }))

  test('Computed value showNoResultMessage compute correctly', () => {
    expect(results.showNoResultMessage).toBeFalsy()
    wrapperResults = shallow(Results, {
      localVue,
      store: {
        state: {
          search: 'mock-search',
          application: 'mock-application'
        },
        getters: {
          numberResults: 0
        }
      },
      stubs: ['router-link', 'router-view'],
      mocks: { $router }
    })
    expect(wrapperResults.vm.showNoResultMessage).toBeTruthy()
  })

  // Here we should expect 6 123 456 not 6,123,456 vue Jest is currently bugging with Intl
  test('Computed value resultNumberSentence forms the right sentence', () => {
    expect(results.resultsNumberSentence).toBe("6,123,456 rÃ©sultats pour \"mock-storedLastFullText\"")
    wrapperResults = shallow(Results, {
      localVue,
      store: {
        state: {
          search: 'mock-search',
          application: 'mock-application'
        },
        getters: {
          numberResults: undefined
        }
      },
      stubs: ['router-link', 'router-view'],
      mocks: { $router }
    })
    expect(wrapperResults.vm.resultsNumberSentence).toBe('')
  })

  test('Computed value resultsAreLoading return the right getter', () => {
    expect(results.resultsAreLoading).toBe(storeMocks.store.state.application.resultsAreLoading)
  })

  test('Computed value serverError return the right getter', () => {
    expect(results.serverError).toBe(storeMocks.store.state.application.serverError)
  })

  test('If only one result, push router to page Etablissement with the right searchId', async () => {
    $router = {
      params: 'mock-params',
      push: jest.fn()
    }

    wrapperResults = shallow(Results, {
      localVue,
      store: {
        state: {
          search: 'mock-search',
          application: 'mock-application'
        },
        getters: {
          numberResults: 1,
          onlyOneResult: true,
          storedResultsEtablissements: [{ name: 'mock-etablissement', siret: 'mock-siret1' }]
        }
      },
      stubs: ['router-link', 'router-view'],
      mocks: { $router }
    })
    wrapperResults.update()
    expect(wrapperResults.vm.$router.push).toHaveBeenCalledWith({ "name": "Etablissement", "params": { "searchId": 'mock-siret1' } })
  })

  test('If more than one result, doest push router', () => {
    wrapperResults = shallow(Results, {
      localVue,
      store: {
        state: {
          search: 'mock-search',
          application: 'mock-application'
        },
        getters: {
          numberResults: 5,
          onlyOneResult: false,
          storedResultsEtablissements: [{ name: 'mock-etablissement', siret: 'mock-siret1' }]
        }
      },
      stubs: ['router-link', 'router-view'],
      mocks: { $router }
    })
    wrapperResults.update()
    expect(wrapperResults.vm.$router.push.mock.calls).toHaveLength(0)
  })

  // TODO: Test when able to test router-link
  test('Clicking on a result goes to this etablissement page')
})

describe('Results.vue snapshot testing', () => {
  // Snapshot testing
  localVue.component('router-link', RouterLinkStub);
  const $router = {
    params: 'mock-params',
    push: jest.fn()
  }
  const wrapperResults = mount(Results, {
    localVue,
    store: {
      state: {
        search: 'mock-search',
        application: 'mock-application',
        route: {
          query: 'mock-query'
        }
      },
      getters: {
        numberResults: 1,
        onlyOneResult: true,
        storedResultsEtablissements: [{ name: 'mock-etablissement', siret: 'mock-siret1' }]
      }
    },
    mocks: { $router }
  })

  test('It match the snapshot', () => {
    expect(wrapperResults.vm.$el).toMatchSnapshot()
  })
})

