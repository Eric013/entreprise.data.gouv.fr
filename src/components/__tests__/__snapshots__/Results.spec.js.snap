// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results.vue snapshot testing It match the snapshot 1`] = `
<div
  class="container"
>
  <h3>
    1 résultats pour "undefined"
  </h3>
   
  <!---->
   
  <ul>
    <li
      class="panel"
    >
      <div>
        <h4
          class="title"
        >
          
        </h4>
         
        <!---->
         
        <p>
          
        </p>
         
        <p>
           
        </p>
      </div>
    </li>
  </ul>
   
  <!---->
   
  <ul
    class="pagination"
    data-v-82963a40=""
  >
    <li
      class="pagesButtons"
      data-v-82963a40=""
    >
      <a
        data-v-82963a40=""
        tabindex="0"
      >
        Precédent
      </a>
    </li>
      
    <li
      class="pagesButtons"
      data-v-82963a40=""
    >
      <a
        data-v-82963a40=""
        tabindex="0"
      >
        Suivant
      </a>
    </li>
  </ul>
</div>
`;
